airbyte-integrations/connectors/source-sf-marketingcloud-singer/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM python:3.7-slim
+
+# Bash is installed for more convenient debugging.
+RUN apt-get update -y && apt-get install -y bash && apt-get install -y gcc && apt-get install -y git && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /airbyte/integration_code
+COPY source_sf_marketingcloud_singer ./source_sf_marketingcloud_singer
+COPY main.py ./
+COPY setup.py ./
+RUN pip install .
+# this line can be removed if data extensions are required
```

Comment:
what is the upside to removing this? I think it makes me a little nervous to edit code via sed.

Reply:
the problem is that including data extensions breaks the Airbyte UI - at least for our exacttarget account. The alternative to using sed is to create a clone of tap-exacttarget, modify it and refer to the clone as the dependency. We thought that using the one liner in the Dockerfile could be more easily removed by others who wanted to get the data extensions.

Reply:
@sherifnada , to expand a bit on @schlattk 's point, there is currently a bug in the Airbyte UI related to very large catalogs. If the data extensions from the Singer tap are included, it triggers this bug. When the bug in the UI is fixed, then the data extensions can be put back at that point.

Reply:
Understood. In that case what we should do is override the `discover` method in source.py and remove the data extension streams. Something like : 

```

def discover(self, logger, config) -> AirbyteCatalog: 
    catalog = super().discover(logger, config)
    catalog.streams = [s for s in catalog.streams if s.name != "ABCXYZ"]
    return catalog
```

this doesn't alter the singer tap itself but just modifies the Airbyte connector shim to exclude that

Reply:
Hi sherif I'm having some problems implementing this as you suggest. currently we have this as our discover method:
`def discover_cmd(self, logger: AirbyteLogger, config_path: str) -> AirbyteCatalog:
        return f"tap-exacttarget --config {config_path} --discover"`